Remove superfluous MediaRecorderErrorEvent interface

dontcallmedom opened the pull request:
fixes #211


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/dontcallmedom/mediacapture-record/pull/212.html" title="Last updated on Jan 20, 2022, 9:47 AM UTC (071eeac)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-record/212/3fd1025...dontcallmedom:071eeac.html" title="Last updated on Jan 20, 2022, 9:47 AM UTC (071eeac)">Diff</a>

dontcallmedom committed:
Remove superfluous MediaRecorderErrorEvent interface

fixes #211

alvestrand commented on MediaRecorder.bs:
```diff
@@ -670,7 +670,7 @@ dictionary BlobEventInit {
 The UA will throw a {{DOMException}} when the error can be detected at the time
 that the call is made. In all other cases the UA will <a>fire an event</a> named
-{{MediaRecorderErrorEvent}}.  If recording has been started and not yet stopped
+{{ErrorEvent}}.  If recording has been started and not yet stopped
```
Is this an "event named ErrorEvent" or "an ErrorEvent"?

alvestrand commented on MediaRecorder.bs:
```diff
@@ -680,68 +680,27 @@ maximum {{Blob}} size that it will support, or the number of
-
-The {{MediaRecorderErrorEvent}} interface is defined for cases when an event is
-raised that was caused by an error.
+## Error eventsMediaRecorderErrorEvent ## {#errorevent-section}
```
Title is mangled.

alvestrand commented on MediaRecorder.bs:
```diff
@@ -680,68 +680,27 @@ maximum {{Blob}} size that it will support, or the number of
-raised that was caused by an error.
+## Error eventsMediaRecorderErrorEvent ## {#errorevent-section}
 
 To <dfn>fire an error event</dfn>
```
Shouldn't this whole section be replaced with a reference to https://html.spec.whatwg.org/multipage/webappapis.html#the-event-handler-processing-algorithm ?

dontcallmedom replied:
I've simplified to make "fire an error event" mean "fire an event with ErrorEvent as constructor" which I think achieves what we need

dontcallmedom committed:
Rely on existing algorithms for errorevent firing

taking into account https://github.com/w3c/mediacapture-record/pull/212#pullrequestreview-857890844 https://github.com/w3c/mediacapture-record/pull/212#pullrequestreview-857887838 https://github.com/w3c/mediacapture-record/pull/212#pullrequestreview-857891255

foolip commented:
Ping reviewers on this. https://chromium-review.googlesource.com/c/chromium/src/+/3379631 is blocked wondering if this is going to land or not.

alvestrand reviewed (approved):
I approve of this simplification.
